test/mavsdk_tests/test_vtol_figure_eight.cpp

```diff
@@ -44,16 +44,17 @@ TEST_CASE("Figure eight execution clockwise", "[vtol]")
 	tester.store_home();
 	const float takeoff_altitude = 20.f;
 	tester.set_takeoff_altitude(takeoff_altitude);
+	std::this_thread::sleep_for(std::chrono::seconds(3));
```

Comment:
Increasing the timeouts for wait_until_altitude, etc seems fine, inserting a random sleep like this seems a bit hacky. 

Do you know what the core problem is? Could it be fixed more fundamentally in set_takeoff_altitude, etc?

Reply:
The problem is i haven't figured out the root cause yet. When i am running it locally, it sometimes failed because it refuses to arm because of "airspeed selector down" preflight check (and the hacky sleep seems to help there). If that passes then the whole test runs just fine. However, in the CI sometimes it just fails the timeout of the wait_until_altitude test. So i think i have two separate problem. For the local, i think maybe there is something wrong in the wait_until_ready function, but i don't know why it takes CI so much longer to reach the altitude.

Reply:
Increasing the timeouts however seems to work better, i have rerun the test now a few times and it seems to pass fine atm, but still would be good to know the reason.

Reply:
I'm going to merge now to cleanup CI for other PRs, but please update again if you get to the root cause?